src/Microsoft.AspNet.OData.Shared/ODataAPIHandler.cs

```diff
@@ -63,130 +65,171 @@ public abstract class ODataAPIHandler<TStructuralType>: IODataAPIHandler where T
         /// <returns>The type of Nested ODataAPIHandler.</returns>
         public abstract IODataAPIHandler GetNestedHandler(TStructuralType parent, string navigationPropertyName);
 
-        /// <summary>
-        /// The parent object.
-        /// </summary>
-        internal TStructuralType ParentObject { get; set; }
-
         /// <summary>
         /// Apply OdataId for a resource with OdataID container.
         /// </summary>
         /// <param name="resource">resource to apply odata id on.</param>
         /// <param name="model">The model.</param>
-        internal virtual void UpdateLinkedObjects(TStructuralType resource, IEdmModel model)
+        /// <param name="apiHandlerFactory">API Handler Factory.</param>
+        public virtual void UpdateLinkedObjects(TStructuralType resource, IEdmModel model, ODataAPIHandlerFactory apiHandlerFactory)
         {
             if (resource != null && model != null)
             {
-                this.ParentObject = resource;
-                CheckAndApplyODataId(resource, model);
+                ApplyHandlerToObject(resource, model, this, apiHandlerFactory);
```

Comment:
Can you explain this change?

Reply:
I did away with the `CheckAndAppluODataId` method since it wasn't achieving what we wanted. It only focused on applying handlers where there is an `ODataIdContainer`. The new method applies handlers to the object and all nested objects